vector/src/main/java/im/vector/app/features/location/LocationSharingService.kt

```diff
@@ -115,53 +120,18 @@ class LocationSharingService : VectorService(), LocationTracker.Callback {
                 }
     }
 
-    private fun scheduleTimer(roomId: String, durationMillis: Long) {
-        Timer()
-                .apply {
-                    schedule(object : TimerTask() {
-                        override fun run() {
-                            stopSharingLocation(roomId)
-                            timers.remove(this@apply)
-                        }
-                    }, durationMillis)
-                }
-                .also {
-                    timers.add(it)
-                }
-    }
-
-    fun stopSharingLocation(roomId: String) {
+    private fun stopSharingLocation(roomId: String) {
         Timber.i("### LocationSharingService.stopSharingLocation for $roomId")
-
-        launchInIO { session ->
-            when (val result = sendStoppedBeaconInfo(session, roomId)) {
-                is UpdateLiveLocationShareResult.Success -> {
-                    synchronized(roomArgsMap) {
-                        val beaconIds = roomArgsMap
-                                .filter { it.value.roomId == roomId }
-                                .map { it.key }
-                        beaconIds.forEach { roomArgsMap.remove(it) }
-
-                        tryToDestroyMe()
-                    }
-                }
-                is UpdateLiveLocationShareResult.Failure -> callback?.onServiceError(result.error)
-                else -> Unit
-            }
-        }
-    }
-
-    private suspend fun sendStoppedBeaconInfo(session: Session, roomId: String): UpdateLiveLocationShareResult? {
-        return session.getRoom(roomId)
-                ?.locationSharingService()
-                ?.stopLiveLocationShare()
+        removeRoomArgs(roomId)
+        tryToDestroyMe()
     }
 
+    @Synchronized
```

Comment:
in the same context as https://github.com/vector-im/element-android/pull/6356/files#r905917814

it would be great to avoid synchronising and make the entry/exit points thread safe, helps reduce the risk of deadlocks or unnecessary thread locking (especially when the main thread is involved :crossed_fingers: )

Reply:
In the `LocationSharingService`, there is the `onLocationUpdate()` which is not called on the Main Thread since we introduced a [`Deboucer` in the `LocationTracker` which runs on a background thread](https://github.com/vector-im/element-android/blob/3dcec85dea47feb823cc8dcc03cf7871af91be45/vector/src/main/java/im/vector/app/features/location/LocationTracker.kt#L67). Do you think I should try to switch back to main thread when notifying callbacks in `LocationTracker` to be able to remove the synchronization ?

Reply:
Okay, thanks. I think I tried the `Flow` approach but I don't remember why I dropped it... I will try to give it another chance.

Reply:
I removed all the synchronization and introduced a `SharedFlow` for the location updates: [353a8a7](https://github.com/vector-im/element-android/pull/6356/commits/353a8a70eba70635c010047912600acfcb7beaf8)